homeassistant/components/shelly/entity.py

```diff
@@ -20,6 +20,31 @@ def temperature_unit(block_info: dict) -> str:
     return TEMP_CELSIUS
 
 
+def shelly_naming(self, block, entity_type: str):
+    """Naming for switch and sensors."""
+    channels = self.wrapper.device.shelly["num_outputs"]
+    if (
+        self.wrapper.model in ["SHSW-21", "SHSW-25"]
+        and self.wrapper.device.settings["mode"] == "roller"
+    ):
+        channels = 1
+    if channels > 1:
+        if block.type == "device":
+            return self.wrapper.name
+        entity_name = self.wrapper.device.settings["relays"][int(block.channel)]["name"]
+        if not entity_name:
+            entity_name = f"{self.wrapper.name} - channel {str(int(block.channel)+1)}"
```

Comment:
I don't like adding the extra `-` sign. This lengthens the names unnecessarily.

Reply:
Easy to remove, but the intent was to separate name and description when the first is composed by more than 1 word.